dev/PushNotifications/PushManager.cpp

```diff
@@ -4,11 +4,97 @@
 #include "pch.h"
 #include "PushManager.h"
 #include "PushManager.g.cpp"
+#include <winrt\windows.networking.pushnotifications.h>
+#include <winerror.h>
+#include <algorithm>
+#include <vector>
+#include <iostream>
+#include <ChannelResult.h>
+#include <winerror.h>
+#include <PushNotificationErrorCodes.h>
+
+using namespace winrt::Windows::Networking::PushNotifications;
 
 namespace winrt::Microsoft::ProjectReunion::implementation
 {
+    std::vector<winrt::guid> PushManager::s_remoteIdList;
+    std::mutex PushManager::s_mutex;
+    std::unique_lock<std::mutex> PushManager::s_lock (s_mutex, std::defer_lock); // Avoid locking during the constructor call
+    inline constexpr std::uint32_t c_maxBackoffSeconds{ 960 };
+
+    bool PushManager::isChannelRequestRetryable(const winrt::hresult& hrException)
+    {
+        switch (hrException)
+        {
+        case HRESULT_FROM_WIN32(ERROR_TIMEOUT):
+        case WnpErrors::WNP_E_NOT_CONNECTED:
+        case WPN_E_OUTSTANDING_CHANNEL_REQUEST:
+        case WnpErrors:: WNP_E_RECONNECTING:
+        case WnpErrors::WNP_E_BIND_USER_BUSY:
+        case RPC_S_SERVER_UNAVAILABLE:
+            return true;
+        default:
+            return false;
+        }
+    }
+
     Windows::Foundation::IAsyncOperationWithProgress<Microsoft::ProjectReunion::ChannelResult, Microsoft::ProjectReunion::ChannelResult> PushManager::CreateChannelAsync(winrt::guid remoteId)
     {
-        throw hresult_not_implemented();
+        winrt::Microsoft::ProjectReunion::ChannelResult channelResult{ nullptr };
+        auto progress{ co_await winrt::get_progress_token() };
+
+        winrt::check_pointer(&remoteId);
+      
+        s_lock.lock();
```

Comment:
The scope of the lock here extends to other API calls like progress and initializing the channelResult. Can you scope it down to the std::find API call only?

Reply:
Ah I see the concern. Yeah will address it accordingly.